tests/utils/test_sampling_utils.py

```diff
@@ -28,3 +30,42 @@ def test_sample_points_on_plane() -> None:
     np.testing.assert_almost_equal(pts_residuals, np.zeros((num_points, 1)))
 
     assert pts.shape == (10, 3)
+
+
+def test_sample_random_directions() -> None:
+    """Check that the sampled directions are uniformly distributed on the sphere, using their mean and std dev."""
+    num_samples = 1000
+
+    direction_list = sampling_utils.sample_random_directions(num_samples)
+    assert len(direction_list) == num_samples
+
+    directions = np.array([direction.point3() for direction in direction_list])
+    mean_direction = np.mean(directions, axis=0)
+    std_direction = np.std(directions, axis=0)
+    std_direction_relative = std_direction / std_direction[0]
+
+    # Increase number of samples for a smaller threshold
+    assert np.allclose(mean_direction, np.zeros(3), atol=1e-1)
+    assert np.allclose(std_direction_relative, np.ones(3), atol=1e-1)
+
+
+def test_sample_kde_directions() -> None:
+    """Check that the sampled directions are distributed according to input kernel density."""
+    num_samples = 1000
+
+    # centered at [1, 1, 1] with variance 0.1
+    mean = np.array([1, 1, 1])
+    cov = np.array([[0.1, 0, 0], [0, 0.1, 0], [0, 0, 0.1]])
+    samples = np.random.multivariate_normal(mean, cov, num_samples)
+    samples_normd = [Unit3(sample) for sample in samples]
+
+    actual_samples = sampling_utils.sample_kde_directions(samples_normd, num_samples)
+    assert len(actual_samples) == num_samples
+
+    actual_samples_array = np.array([sample.point3() for sample in actual_samples])
```

Comment:
Maybe would be helpful to add an inline comment like
`# Compute empirical sample mean and covariance and verify matches specified distribution.`

Reply:
that is what the test description means, but I can choose this exact wording if you prefer that :)